semcore/d3-chart/src/RadialTree.tsx

```diff
@@ -0,0 +1,768 @@
+import createElement from './createElement';
+import uniqueIDEnhancement from '@semcore/utils/lib/uniqueID';
+import { transition } from 'd3-transition';
+import style from './style/radial-tree.shadow.css';
+import { shade } from '@semcore/utils/lib/color';
+
+const memoize = <T extends (...args: (string | number)[]) => any>(func: T): T => {
+  const results: { [cacheKey: string]: ReturnType<T> } = {};
+  return ((...args) => {
+    const argsKey = args.join(',');
+    if (!results[argsKey]) {
+      results[argsKey] = func(...args);
+    }
+    return results[argsKey];
+  }) as T;
+};
+
+const measureText = memoize((text: string, fontSize: number) => {
+  const span = document.createElement('span');
+  span.style.fontSize = `${fontSize}px`;
+  for (const line of text.split('\n')) {
+    span.append(document.createTextNode(line));
+    span.append(document.createElement('br'));
+  }
+  span.style.display = 'inline-block';
+  document.body.append(span);
+  const textWidth = span.offsetWidth;
+  span.remove();
+  return textWidth;
+});
+
+const baseAngle = -Math.PI / 2; // The top vertical line
+
+type RadianData = {
+  /**
+   * Text label on the end of radian.
+   */
+  label: string;
+  /**
+   * Radian key. Radian, which `key` is equal to `activeKey` is displayed as active.
+   */
+  key: string;
+  /**
+   * Size of the cap (filled circle on the end of the radian).
+   * @default 16
+   */
+  capSize?: number;
+  /**
+   * Icon, that displayed in the cap of the active radian (usage example `icon: UserM`).
+   */
+  icon?: React.FC;
+  /**
+   * Color of radian's line, cap and label.
+   * @default #000
+   */
+  color?: string;
+  /**
+   * Color of icon.
+   * @default #fff
+   */
+  iconColor?: string;
+  /**
+   * Size of icon.
+   * @default 16
+   */
+  iconSize?: number;
+};
+
+export interface IRadialTreeProps {
+  /**
+   * List of radians. `{ label: string; key: string; capSize?: number; icon?: React.FC; color?: string; iconColor?: string; iconSize?: number; }`
+   */
+  data?: RadianData[];
+  /**
+   * Angel (in rads) that rotates chart. 0 means that first radian is exactly on top vertical line.
+   * @default Math.PI / 5
+   */
+  angleOffset?: number;
+  /**
+   * Duration of appear and update animation.
+   * @default 300
+   */
+  duration?: number;
+  /**
+   * Gap between the chart center and radians start point.
+   * @default 50
+   */
+  centralMargin?: number;
+  /**
+   * Gap around text label.
+   * @default 2
+   */
+  labelMargin?: number;
+  /**
+   * Color of all radian's line, cap and label.
+   * @default #000
+   */
+  color?: string;
+  /**
+   * Size of the cap (filled circle on the end of the radian).
+   * @default 16
+   */
+  capSize?: number;
+  /**
+   * Icon, that displayed in the cap of the active radian (usage example `icon: UserM`).
+   */
+  icon?: React.FC;
+  /**
+   * Color of icon.
+   * @default #fff
+   */
+  iconColor?: string;
```

Comment:
How I understand this property I can set in Plot `<Plot data={{iconColor: ...}}>`. I this bad idea, because you RadialTree has Icon component and you should set property for icon in component  RadialTree.Icon. I think this same should do for prop `icon, iconSize`.